my_server1.py

```diff
@@ -0,0 +1,39 @@
+from http.server import HTTPServer, BaseHTTPRequestHandler
+from urllib.parse import parse_qs
+import json
+from file_manager import FileManager
+
+
+class MessageHandler(BaseHTTPRequestHandler):
+    def do_POST(self):
+        # 1. How long was the message? (Use the Content-Length header.)
+        length = int(self.headers.get('Content-length', 0))
+
+        # 2. Read the correct amount of data from the request.
+        data = self.rfile.read(length).decode()
+
+        # 3. Extract the "message" field from the request data.
+        menu_number = parse_qs(data)["menu_number"][0]
+
+        # Send the "message" field back as the response.
+        self.send_response(200)
+        self.send_header('Content-type', 'text/plain; charset=utf-8')
+        self.end_headers()
+      
+
+        data_string = ""
+        with FileManager('info_about_lunch.txt', 'r') as lunch_details_file:
```

Comment:
You have created your own context manager wrapper (which is good for learning) but open returns context manager itself. So you don't need to create a wrapper.
```py
with open('lunch_menu.txt', 'r') as lunch_details:
     #do stuff

```

Please give an example scenario where we can need a context manager.

Reply:
sir we can use it for Database connection
or we our application is connected to many data sources and we need to close the connections.